custodia/httpd/server.py

```diff
@@ -436,25 +464,39 @@ class HTTPServer(object):
 
     def __init__(self, srvurl, config):
         url = urlparse(srvurl)
-        address = unquote(url.netloc)
         if url.scheme == 'http+unix':
             # Unix socket
+            address = None
+            if sd_listen_fds is not None:
```

Comment:
This should be done regardless of the scheme used, and placed before url.scheme checking.
If the scheme then does not match the provided socket error out.
If proper support to check the socket type is not available (better if we can do that in a forward compatible way) then it is o to assume http+unix.